source/common/http/http2/codec_impl.cc

```diff
@@ -1791,9 +1808,10 @@ ConnectionImpl::Http2Callbacks::Http2Callbacks() {
 
   nghttp2_session_callbacks_set_on_frame_recv_callback(
       callbacks_, [](nghttp2_session*, const nghttp2_frame* frame, void* user_data) -> int {
-        auto status = static_cast<ConnectionImpl*>(user_data)->onFrameReceived(frame);
-        return static_cast<ConnectionImpl*>(user_data)->setAndCheckCodecCallbackStatus(
-            std::move(status));
+        auto* conn = static_cast<ConnectionImpl*>(user_data);
+        RELEASE_ASSERT(!conn->skipCallbackVisitor(), "Unexpected use of nghttp2 callback!");
+        auto status = conn->onFrameReceived(frame);
```

Comment:
nit: can we use the full type here (and elsewhere where the return value is a Status)

Reply:
Done, thanks.